src/cpp.cc

```diff
@@ -1537,6 +1537,18 @@ void Printer::Cpp::print_seq_init(const AST &ast) {
            << "\"_checkpointing_log.txt\";" << endl;
     stream << indent() << "logfile_path = opts.checkpoint_out_path / "
            << "logfile_name;" << endl << endl;
+    if (ast.checkpoint->cyk) {
+      stream << indent() << "std::string cyk_archive = "
+             << "file_prefix + \"_cyk_indices\";" << endl;
+      stream << indent() << "out_cyk_path = opts.checkpoint_out_path / "
+             << "cyk_archive;" << endl;
+      stream << indent() << "tmp_out_cyk_path = opts.checkpoint_out_path / "
+             << "(cyk_archive + \"_new\");" << endl << endl;
+      for (size_t i = 0; i < ast.grammar()->axiom->tracks(); i++) {
```

Comment:
it is not guaranteed that any NT has the same number of tracks as the axiom!

Reply:
Ah, good to know. It's kind of hard to figure all the differences between the different tables just from reading the source code, especially since you've also mentioned that the `CONST` table can also have running indices. I will have another look at the whole mechanism and test some more (multi track) programs to hopefully be able to spot all the edge cases.